File: WireCutting/WireCutting.c
#include <mraa.h>
#include <signal.h>
#include <stdlib.h>

sig_atomic_t volatile run_flag = 1;

void sig_handler(int sig)
{
	if (sig == SIGINT)
		run_flag = 0;
}

void BOOM(int value)
{
	value = 0;
}

int main()
{
	mraa_aio_context adc_a0;
	uint16_t adc_value = 0;
	float adc_value_float = 0.0;
	mraa_result_t r = MRAA_SUCCESS;
	adc_a0 = mraa_aio_init(0);
	if (adc_a0 == NULL) {
		return 1;
	}
	srand(time(NULL));
	int rando = rand() %5;
	//signal(SIGINT, sig_handler);

	/*while (run_flag) {
		adc_value = mraa_aio_read(adc_a0);
	        adc_value_float = mraa_aio_read_float(adc_a0);
	        fprintf(stdout, "V: %f\n", adc_value_float);
	}*/

	adc_value = mraa_aio_read(adc_a0);
	adc_value_float = mraa_aio_read_float(adc_a0);
	
	if (rando == 0)
	{
		fprintf(stdout, "Cut Red/White wire!\n");
		while (adc_value_float > 0.313f && adc_value_float < 0.319f)
		{
			adc_value = mraa_aio_read(adc_a0);
	        	adc_value_float = mraa_aio_read_float(adc_a0);
			if (adc_value_float < 0.313f || adc_value_float > 0.319f)
			{
				if (adc_value_float > 0.458f && adc_value_float < 0.462f)
				{
					fprintf(stdout, "SUCCESS!\n");
				}
				else
				{
					fprintf(stdout, "BOOM!\n");
				}
			}	
		}
	}
	else if (rando == 1)
	{
		fprintf(stdout, "Cut Brown/Green wire!\n");
		while (adc_value_float > 0.313f && adc_value_float < 0.319f)
		{
			adc_value = mraa_aio_read(adc_a0);
	        	adc_value_float = mraa_aio_read_float(adc_a0);
			if (adc_value_float < 0.313f || adc_value_float > 0.319f)
			{
				if (adc_value_float > 0.368f && adc_value_float < 0.372f)
				{
					fprintf(stdout, "SUCCESS!\n");
				}
				else
				{
					fprintf(stdout, "BOOM!\n");
				}
			}			
		}
	}
	else if (rando == 2)
	{
		fprintf(stdout, "Cut Brown/Blue wire!\n");
		while (adc_value_float > 0.313f && adc_value_float < 0.319f)
		{
			adc_value = mraa_aio_read(adc_a0);
	        	adc_value_float = mraa_aio_read_float(adc_a0);
			if (adc_value_float < 0.313f || adc_value_float > 0.319f)
			{
				if (adc_value_float > 0.347f && adc_value_float < 0.352f)
				{
					fprintf(stdout, "SUCCESS!\n");
				}
				else
				{
					fprintf(stdout, "BOOM!\n");
				}
			}			
		}
	}
	else if (rando == 3)
	{
		fprintf(stdout, "Cut Red/Red wire!\n");
		while (adc_value_float > 0.313f && adc_value_float < 0.319f)
		{
			adc_value = mraa_aio_read(adc_a0);
	        	adc_value_float = mraa_aio_read_float(adc_a0);
			if (adc_value_float < 0.313f || adc_value_float > 0.319f)
			{
				if (adc_value_float > 0.340f && adc_value_float < 0.344f)
				{
					fprintf(stdout, "SUCCESS!\n");
				}
				else
				{
					fprintf(stdout, "BOOM!\n");
				}
			}			
		}
	}
	else if (rando == 4)
	{
		fprintf(stdout, "Cut Black/Black wire!\n");
		while (adc_value_float > 0.313f && adc_value_float < 0.319f)
		{
			adc_value = mraa_aio_read(adc_a0);
	        	adc_value_float = mraa_aio_read_float(adc_a0);
			if (adc_value_float < 0.313f || adc_value_float > 0.319f)
			{
				if (adc_value_float > 0.335f && adc_value_float < 0.3395f)
				{
					fprintf(stdout, "SUCCESS!\n");
				}
				else
				{
					fprintf(stdout, "BOOM!\n");
				}
			}			
		}
	}	
	
	r = mraa_aio_close(adc_a0);
	if (r != MRAA_SUCCESS) {
	        mraa_result_print(r);
	}
	return r;
}

File: WireButton/WireCutting/WireCutting.py
import mraa
import random
import time

class WireCutting:
	def __init__(self, diff):
		#random.seed()
		#self.RANDGEN = random.randint(0,4)
		self.difficulty = diff
		#flags for cut wire

	def getValues(self):
		self.a0 = mraa.Aio(0)
		self.a1 = mraa.Aio(1)
		self.a2 = mraa.Aio(2)
		self.a3 = mraa.Aio(3)
		self.a4 = mraa.Aio(4)


	def startGame(self):



		if RANDGEN == 0:
			print "Cut the RED + RED wire!"
			while True:
				if self.a0.read() < 160:
					print "SUCCESS"
					return(1)
				elif self.a1.read() < 160 or self.a2.read() < 160 or self.a3.read() < 160 or self.a4.read() < 160:
					print "FAILURE"
					return(0)
		elif RANDGEN == 1:
			print "Cut the BROWN + GREEN wire!"
			while True:
				self.getValues()
				if self.a1.read() < 160:
					print "SUCCESS"
					return(1)
				elif self.a0.read() < 160 or self.a2.read() < 160 or self.a3.read() < 160 or self.a4.read() < 160:
					print "FAILURE"
					return(0)
		elif RANDGEN == 2:
			print "Cut the BLACK + BLACK wire!"
			while True:
				self.getValues()
				if self.a2.read() < 160:
					print "SUCCESS"
					return(1)
				elif self.a1.read() < 160 or self.a0.read() < 160 or self.a3.read() < 160 or self.a4.read() < 160:
					print "FAILURE"
					return(0)
		elif RANDGEN == 3:
			print "Cut the BROWN + BLUE wire!"
			while True:
				self.getValues()
				if self.a3.read() < 160:
					print "SUCCESS"
					return(1)
				elif self.a1.read() < 160 or self.a2.read() < 160 or self.a0.read() < 160 or self.a4.read() < 160:
					print "FAILURE"
					return(0)
		elif RANDGEN == 4:
			print "Cut the RED + WHITE wire!"
			while True:
				self.getValues()
				if self.a4.read() < 160:
					print "SUCCESS"
					return(1)
				elif self.a1.read() < 160 or self.a2.read() < 160 or self.a3.read() < 160 or self.a0.read() < 160:
					print "FAILURE"
					return(0)

if __name__ == '__main__':
	g = WireCutting()
	#g.startGame(1)

	t = WireCutting()
	while True:
		t.getValues()
		print t.a0.read()
		print t.a1.read()
		print t.a2.read()
		print t.a3.read()
		print t.a4.read()